CMSgov/mint-app
# [EASI-4309] Front End work for the Dates Changed Notification

# EASI-4309


## Changes and Description

- Add `DatesChanged` component
- Update `IndividualNotification` component to accommodate the `DatesChanged` component
- Add Dates Changed Notification settings in the `Settings` page
- Updated `GetNotification` and `GetNotificationSettings` gql queries
- Add new copy to translation files
- Updated Cypress test to account for this new notification type



## How to test this change

1. Enable email and in-app Dates Changed notifications in the Notification settings page
2. Verify that preferences persist
3. Update dates for any model plan
4. Return to Notification page and verify that a new "Dates Changed" notification popped up
5. Hit the dropdown and verify the dates changed matches and the design matches figma
6. Unsubscribe via email link `http://localhost:3005/notifications/settings?unsubscribe_email=DATES_CHANGED` 
7. Verify that unsubscribing actually works
8. Verify that tests pass




## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
